src/main/java/uk/ac/cam/cl/dtg/isaac/api/PagesFacade.java

```diff
@@ -388,40 +389,72 @@ public final Response getQuestionList(@Context final Request request,
-                    fieldsToMatch.get(TOPICS_FIELDNAME),
-                    fieldsToMatch.get(BOOKS_FIELDNAME),
-                    fieldsToMatch.get(LEVEL_FIELDNAME),
-                    fieldsToMatch.get(STAGE_FIELDNAME),
-                    fieldsToMatch.get(DIFFICULTY_FIELDNAME),
-                    fieldsToMatch.get(EXAM_BOARD_FIELDNAME),
-                    fieldsToMatch.getOrDefault(TYPE_FIELDNAME, Set.of(QUESTION_TYPE)),
-                    newStartIndex,
-                    newLimit,
-                    showNoFilterContent
-            );
-
-            ResultsWrapper<ContentSummaryDTO> summarizedContent = new ResultsWrapper<>(
-                    this.extractContentSummaryFromList(c.getResults()),
-                    c.getTotalResults());
-
-            return Response.ok(summarizedContent).tag(etag)
-                    .cacheControl(getCacheControl(NUMBER_SECONDS_IN_ONE_HOUR, true))
-                    .build();
+        List<ContentSummaryDTO> combinedResults = new ArrayList<>();
+        int nextSearchStartIndex = newStartIndex;
+        Long totalResults = 0L;
+
+        List<ContentSummaryDTO> summarizedResults;
+
+        for (int iterationLimit = 5; iterationLimit > 0 && (newLimit < 0 || combinedResults.size() < newLimit); iterationLimit--) {
+            try {
+                ResultsWrapper<ContentDTO> c;
+                c = contentManager.searchForContent(
+                        validatedSearchString,
+                        fieldsToMatch.get(ID_FIELDNAME),
+                        fieldsToMatch.get(TAGS_FIELDNAME),
+                        fieldsToMatch.get(SUBJECTS_FIELDNAME),
+                        fieldsToMatch.get(FIELDS_FIELDNAME),
+                        fieldsToMatch.get(TOPICS_FIELDNAME),
+                        fieldsToMatch.get(BOOKS_FIELDNAME),
+                        fieldsToMatch.get(LEVEL_FIELDNAME),
+                        fieldsToMatch.get(STAGE_FIELDNAME),
+                        fieldsToMatch.get(DIFFICULTY_FIELDNAME),
+                        fieldsToMatch.get(EXAM_BOARD_FIELDNAME),
+                        fieldsToMatch.getOrDefault(TYPE_FIELDNAME, Set.of(QUESTION_TYPE)),
+                        nextSearchStartIndex,
+                        newLimit,
+                        showNoFilterContent
+                );
 
-        } catch (ContentManagerException e1) {
-            SegueErrorResponse error = new SegueErrorResponse(Status.NOT_FOUND,
-                    "Error locating the content requested", e1);
-            log.error(error.getErrorMessage(), e1);
-            return error.toResponse();
+                summarizedResults = extractContentSummaryFromList(c.getResults());;
+
+                if (summarizedResults.isEmpty()) {
+                    break;
+                }
+
+                nextSearchStartIndex += summarizedResults.size(); // add the total number of returned results, including any we will filter out
```

Comment:
This can do the wrong thing. Say there are 65 questions matching the search and we have a page size of 30. If the user has answered 5 of the first page, we will load the next 30 and increment `nextSearchStartIndex` from 30 to 60. But we will then throw away 25 of these, since we only need 5 to fill the list. The next request will start at an index of 60, and those middle 25 will never be able to be loaded.